### PR TITLE
Reworking model models after the hydration changes.

### PR DESCRIPTION
**Pull request by bcdurak**
## Describe changes

I converted model-related response models to work with our new modeling paradigm after the hydration changes.

## Pre-requisites
Please ensure you have done the following:
- [X] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [X] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/stacks-and-components/component-guide) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.
- [X] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/model/artifact_config.py [hunk 1]
```diff
@@ -172,7 +177,8 @@ def _link_to_model_version(
             if self.overwrite:
                 # delete all model version artifact links by name
                 logger.warning(
-                    f"Existing artifact link(s) `{artifact_name}` found and will be deleted."
+                    f"Existing artifact link(s) `{artifact_name}` found and "
+                    f"will be deleted."
```
Comment (avishniakov) [inline src/zenml/model/artifact_config.py:181]: remove `f`

File: src/zenml/model/artifact_config.py [hunk 2]
```diff
@@ -181,7 +187,8 @@ def _link_to_model_version(
                 )
             else:
                 logger.info(
-                    f"Artifact link `{artifact_name}` already exists, adding new version."
+                    f"Artifact link `{artifact_name}` already exists, adding "
+                    f"new version."
```
Comment (avishniakov) [inline src/zenml/model/artifact_config.py:191]: remove `f`

File: src/zenml/zen_stores/schemas/model_schemas.py [hunk 3]
```diff
@@ -139,32 +143,40 @@ def to_model(
...
             created=self.created,
             updated=self.updated,
-            license=self.license,
-            description=self.description,
-            audience=self.audience,
-            use_cases=self.use_cases,
-            limitations=self.limitations,
-            trade_offs=self.trade_offs,
-            ethics=self.ethics,
             tags=tags,
```
Comment (avishniakov) [inline src/zenml/zen_stores/schemas/model_schemas.py:164]: Tags are TagsResponseModel list, does it belong here?

**Review by avishniakov — CHANGES_REQUESTED** [review]
Left a few comments, now 100 integration tests failed - will do a better review once fixed.

File: src/zenml/model/artifact_config.py (hunk 1 above)
Comment (bcdurak) [reply src/zenml/model/artifact_config.py]: Fixed it!

File: src/zenml/models/v2/core/model.py [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+    remove_tags: Optional[List[str]]
+
+
+# ------------------ Response Model ------------------
+
+
+class ModelResponseBody(WorkspaceScopedResponseBody):
+    """Response body for models."""
+
+    tags: List["TagResponseModel"] = Field(
```
Comment (fa9r) [inline src/zenml/models/v2/core/model.py:101]: should tags be in metadata or body? Probably @avishniakov or @htahir1 know this best?

For artifacts I added them to metadata, we should probably keep that consistent

File: src/zenml/models/v2/core/model.py [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+        )
+        ret = [
+            mv.to_model_version(suppress_class_validation_warnings=True)
+            for mv in model_versions.items
+        ]
+        for i in range(2, model_versions.total_pages + 1):
+            ret += [
+                mv.to_model_version(suppress_class_validation_warnings=True)
+                for mv in model_versions.items
+            ]
```
Comment (fa9r) [inline src/zenml/models/v2/core/model.py:188]: This doesn't seem right, you iterate over `for mv in model_versions.items` N times here, I'm guessing the goal is to get all pages of model versions? If so, better use `pagination_utils.depaginate()` if you want to get ALL items

File: src/zenml/models/v2/core/model.py [hunk 6]
```diff
@@ -0,0 +1,314 @@
...
+
+        return ret
+
+    # Body and metadata properties
+    @property
```
Comment (fa9r) [inline src/zenml/models/v2/core/model.py:193]: redundant properties!!1!!11

File: src/zenml/models/v2/core/model_version_pipeline_run.py [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+    name: Optional[str] = Field(
+        description="The name of the pipeline run inside model version.",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    pipeline_run: UUID
+    model: UUID
+    model_version: UUID
+
+
+class ModelVersionPipelineRunResponse(
```
Comment (fa9r) [inline src/zenml/models/v2/core/model_version_pipeline_run.py:64]: This has no metadata - will that not break your sanity check logic?

**Review by fa9r — APPROVED** [review]
I think we should unify where tags go, otherwise LGTM

File: src/zenml/models/v2/core/model.py (hunk 4 above)
Comment (htahir1) [reply src/zenml/models/v2/core/model.py:101]: Wouldnt it make sense to have them outside? I lean towards that

Comment (avishniakov) [reply src/zenml/models/v2/core/model.py:101]: How I understand this whole hydration concept: what is in the body (hydrate=False) belongs to listers, and what is in metadata (hydrate=True) belongs to getters. So Tags belong to the body, cause it is used in lists by the frontend.

**Review by avishniakov — COMMENTED** [review]
I feel tags belong to the body, rest is good.

**Comment by bcdurak** [comment]
Unfortunately, due to the recent changes to the model models, this branch is cursed with countless merge conflicts. Instead of trying to resolve all of these conflicts, I will close this PR without merging and create a brand new one with the same goal. I will take all of your comments into consideration in the new PR as I move on to the new branch.

**Pull request closed without merging**